### DISCUSSION TITLE
Historical run results lifespan

### DISCUSSION THREAD

**Discussion by parky128** [sorry-cypress/sorry-cypress#405] [General]
Are there any configuration options with how long to keep run results for in the database or is that totally down to us to control?

I can foresee historical runs filling up disk space and wondered what can be done to auto delete these after a certain period? 

I did see a [deleteRun](https://github.com/sorry-cypress/sorry-cypress/blob/0ec6ea72759be59a98a28b1fce75ee46564e187f/packages/api/src/schema/schema.graphql#L21) Mutation in the graphql schema and so if nothing does exist for auto cleanup\delete of old runs, I wondered about writing something that could maybe delete runs after say a Pull Request has been merged, or something similar

**Comment by agoldis** [comment]
> Are there any configuration options with how long to keep run results for in the database 

no, it's totally up to you

> I can foresee historical runs filling up disk space and wondered what can be done to auto delete these after a certain period?

Yes, you can just delete those

> I did see a deleteRun Mutation in the graphql schema

There're some graphql endpoint that are capable of removing runs in bulk, but they never were exposed via UI.
